src/nvim/mbyte.c

```diff
@@ -1092,9 +1090,52 @@ bool utf_iscomposing(int c)
   return intable(combining, ARRAY_SIZE(combining), c);
 }
 
+#ifdef __SSE2__
+
+# include <emmintrin.h>
+
+// Return true for characters that can be displayed in a normal way.
+// Only for characters of 0x100 and above!
+bool utf_printable(int c)
+  FUNC_ATTR_CONST
+{
+  if (c == 0x70f) {
+    return false;
+  }
+  if (c < 0x180b || c > 0xffff) {
+    return true;
+  }
```

Comment:
This seems to be faster as the result of the `c != 0x070f` comparison is used directly as the return value:
```suggestion
  if (c < 0x180b || c > 0xffff) {
    return c != 0x070f;
  }
```
Also, codepoints above `0xffff` should be less common than codepoints below `0xffff`, so maybe it will be faster to remove this `c > 0xffff` check and instead let the SIMD comparisons below cover it?
```suggestion
  if (c < 0x180b) {
    return c != 0x070f;
  }
```

Reply:
I used 16 bit comparisons to fit 8 values in __m128i, so the value of `c` needs to be between 0 and FFFF.

Reply:
Oh right.